src/components/Geo/Geo.tsx

```diff
@@ -1,14 +1,93 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import { Helmet } from 'react-helmet'
+import { SpeciesSelect } from './SpeciesSelect'
 import { MapPlot } from './MapPlot'
+import { TimePlot } from './TimePlot'
 import { SelectionInfo } from './SelectionInfo'
 import { RunData } from './types'
 import { helpIcon } from 'common'
+import * as d3 from 'd3'
+import rdrpPosTsv from './rdrp_pos.tsv'
+import { DateTime } from 'luxon'
+
+async function retrieveAllPossibleRows(
+    inputFile: any,
+    setAllPossibleRows: React.Dispatch<React.SetStateAction<RunData[] | undefined>>
+) {
+    let rows = ((await d3.tsv(inputFile)) as object) as RunData[]
+    setAllPossibleRows(rows)
+}
+
+/**
+ * This component:
+ * - Begins by reading in all the possible rows contained in the rdrpPosTsv file
+ * - Gets all the unique species to pass to the multi-select dropdown
+ * - Creates an object of release dates (months) to number of species for all rows
+ * - Returns the components of the Geo page
+ */
 
 export const Geo = () => {
+    const [allPossibleRows, setAllPossibleRows] = React.useState<RunData[]>()
+    const [allUniqueSpecies, setAllUniqueSpecies] = React.useState<string[]>()
+    const [allRowsTimePlot, setAllRowsTimePlot] = React.useState<{ [key: string]: number }>()
+
     const [selectedPoints, setSelectedPoints] = React.useState<RunData[]>()
+    const [selectedSpecies, setSelectedSpecies] = React.useState<String[]>()
+    const [selectedRows, setSelectedRows] = React.useState<RunData[]>()
     const [isCollapsed, setIsCollapsed] = React.useState<boolean>(false)
 
+    // Retrieve all possible rows from the rdrpPosTsv file
+    if (allPossibleRows === undefined) retrieveAllPossibleRows(rdrpPosTsv, setAllPossibleRows)
+    // From all of the possible rows, distill the unique species to be passed into the SpeciesSelect component
+    if (allPossibleRows && allUniqueSpecies === undefined)
+        setAllUniqueSpecies([...new Set(allPossibleRows.map((d) => d.scientific_name))].sort())
+    // Create an object with key:value pairs corresponding to all months from earliest to latest release date and the number of species' release dates per month
+    if (allPossibleRows && allRowsTimePlot === undefined) {
+        // Create an array of months spanning the earliest one within all rows on to the latest
+        let monthsArray: string[] = []
+        const uniqueMonths = [...new Set(allPossibleRows.map((row) => row.release_date))].sort()
+        let earliestMonth = DateTime.fromISO(uniqueMonths.slice(0)[0].split(' ')[0])
+        const latestMonth = DateTime.fromISO(uniqueMonths.slice(-1)[0].split(' ')[0])
+
+        if (latestMonth < earliestMonth) {
+            throw 'Latest date must be greater than earliest.'
+        }
```

Comment:
Yes it's a safeguard but I think it's unnecessary so will remove.

Reply:
I actually liked it, there's not enough tests in code these days and this is something we will see during development and otherwise is essentially free (cpu time wise).

Reply:
I kept it in.  Can't hurt.